src/token-processor/util/helpers.ts

```diff
@@ -30,3 +31,34 @@ export function waiter<T = void>(): Waiter<T> {
   };
   return Object.assign(promise, completer);
 }
+
+/**
+ * Parses a `Retry-After` HTTP header from an undici 429 `ResponseStatusCodeError` error so we can
+ * determine when we can try calling the same host again looking for metadata.
+ * @param error - Original ResponseStatusCodeError
+ * @returns retry-after value in seconds since now
+ */
+export function parseRetryAfterResponseHeader(
+  error: errors.ResponseStatusCodeError
+): number | undefined {
+  if (error.statusCode != 429 || !error.headers || !('retry-after' in error.headers)) {
+    return;
+  }
+  const header = error.headers['retry-after'];
+  if (!header) return;
+  const wrappedValue = Array.isArray(header) ? header : [header];
+
+  // Return the first valid header value we find.
+  for (const value of wrappedValue) {
+    // Numerical values e.g. `Retry-After: 120`
+    const nval = Number(value);
+    if (Number.isFinite(nval) && nval > 0) {
+      return nval;
+    }
+    // HTTP Date values e.g. `Retry-After: Wed, 21 Oct 2015 07:28:00 GMT`
+    const retryDateMS = Date.parse(value);
+    if (Number.isNaN(retryDateMS)) return;
```

Comment:
What is the intention with checking if a `Date` instance is NaN here?

Reply:
As far as I could see, that guarded against invalid date strings... maybe there's a better way to do it, though? Open to ideas if you have any

Reply:
Ahh I see, I didn't realized `Date.parse()` returned `number` rather than `date`. Thanks!